lib/cext/preprocess.rb

```diff
@@ -834,7 +834,7 @@ def patch(file, contents, directory)
                 directory.end_with?(File.join(patched_file[:gem], 'ext', patched_file[:ext_dir]))
               else
                 regexp = /^#{Regexp.escape(patched_file[:gem])}\b/
-                directory.split('/').last(2).any? { |part| part =~ regexp }
+                directory.split('/').last(2).any? { |part| part =~ regexp } || file.split('/').last(2).any? { |part| part =~ regexp }
```

Comment:
Maybe we should just match with `File.expand_path(file).end_with?("/#{key}")` but that would require iterating PATCHED_FILES's keys (likely OK here).
We could also consistently use a fixed number of path components like 2, then we can do a simple lookup.

Reply:
I agree this needs to be cleaned up. The basic assumption made when this pattern matching system was set hasn't held. The naming conventions used are much more varied. But, let's handle that in a separate PR if you don't mind.

Reply:
Sure.